cli/config/config.go

```diff
@@ -0,0 +1,112 @@
+					Flags: []cli.Flag{
+						cli.BoolFlag{Name: "privnet", Usage: "private network configuration"},
+						cli.BoolFlag{Name: "mainnet", Usage: "mainnet network configuration"},
+						cli.BoolFlag{Name: "testnet", Usage: "testnet network configuration"},
+						cli.BoolFlag{Name: "unit_testnet", Usage: "unit test network configuration"},
+						cli.BoolFlag{Name: "mainnet_neofs", Usage: "mainnet NeoFS network configuration"},
+						cli.BoolFlag{Name: "testnet_neofs", Usage: "testnet NeoFS network configuration"},
+						cli.BoolFlag{Name: "docker", Usage: "Docker configuration"},
+						cli.BoolFlag{Name: "all", Usage: "all networks configurations"},
+					},
+				},
+			},
+		},
+	}
+}
+
+func configGenerate(ctx *cli.Context) error {
+	if err := cmdargs.EnsureNone(ctx); err != nil {
+		return err
+	}
+
+	var filesToGenerate map[string]config.Config
+
+	if ctx.Bool("all") {
+		filesToGenerate = map[string]config.Config{
+			"protocol.mainnet.yml":               MainNetConfig,
+			"protocol.mainnet.neofs.yml":         NeoFSMainNetConfig,
+			"protocol.testnet.yml":               TestNetConfig,
+			"protocol.testnet.neofs.yml":         NeoFSTestNetConfig,
+			"protocol.unit_testnet.yml":          UnitTestNetConfig,
+			"protocol.unit_testnet.single.yml":   UnitTestNetSingleConfig,
+			"protocol.privnet.docker.one.yml":    PrivNetDockerOneConfig,
+			"protocol.privnet.docker.two.yml":    PrivNetDockerTwoConfig,
+			"protocol.privnet.docker.three.yml":  PrivNetDockerThreeConfig,
+			"protocol.privnet.docker.four.yml":   PrivNetDockerFourConfig,
+			"protocol.privnet.docker.single.yml": PrivNetDockerSingleConfig,
+			"protocol.privnet.yml":               PrivNetConfig,
+		}
+	} else {
+		if ctx.Bool("privnet") {
+			filesToGenerate["protocol.privnet.yml"] = PrivNetConfig
+		}
+		if ctx.Bool("mainnet") {
+			filesToGenerate["protocol.mainnet.yml"] = MainNetConfig
+		}
+		if ctx.Bool("testnet") {
+			filesToGenerate["protocol.testnet.yml"] = TestNetConfig
+		}
+		if ctx.Bool("unit_testnet") {
+			filesToGenerate["protocol.unit_testnet.yml"] = UnitTestNetConfig
+			filesToGenerate["protocol.unit_testnet.single.yml"] = UnitTestNetSingleConfig
+		}
+		if ctx.Bool("mainnet_neofs") {
+			filesToGenerate["protocol.mainnet.neofs.yml"] = NeoFSMainNetConfig
+		}
+		if ctx.Bool("testnet_neofs") {
+			filesToGenerate["protocol.testnet.neofs.yml"] = NeoFSTestNetConfig
+		}
+		if ctx.Bool("docker") {
+			filesToGenerate["protocol.privnet.docker.one.yml"] = PrivNetDockerOneConfig
+			filesToGenerate["protocol.privnet.docker.two.yml"] = PrivNetDockerTwoConfig
+			filesToGenerate["protocol.privnet.docker.three.yml"] = PrivNetDockerThreeConfig
+			filesToGenerate["protocol.privnet.docker.four.yml"] = PrivNetDockerFourConfig
+			filesToGenerate["protocol.privnet.docker.single.yml"] = PrivNetDockerSingleConfig
+		}
+	}
+
+	for fileName, cfg := range filesToGenerate {
+		if err := writeConfigToFile(cfg, filepath.Join("config", fileName)); err != nil {
```

Comment:
For this you need a separate argument: target path to write config files to.

Reply:
We have `--config-path` flag that is shared between all other commands, we need to reuse it.

Reply:
reuse leads to import cycles and the `options.Config` flag description seems a little bit confusing "path to directory with per-network configuration files (may be overridden by --config-file option for the configuration file)"

Reply:
Import cycles may be solved by moving it to a separate package. But the second argument is accepted, we need its own usage text for this flag, thus, let's create a new `--config-path` flag for this package.